src/components/PostDetails.tsx

```diff
@@ -1,106 +1,130 @@
+  const handleDeleteComment = async (commentId: number) => {
+    try {
+      setComments(comments.filter(comment => comment.id !== commentId));
+      await deleteComment(commentId);
+    } catch {
+      setComments(comments);
+    }
+  };
 
-            <div className="message-body" data-cy="CommentBody">
-              Some comment
-            </div>
-          </article>
+  const handleCommentCreate = async (comment: Omit<Comment, 'id'>) => {
+    try {
+      const commentDataFromServer = await postComment(comment);
 
-          <article className="message is-small" data-cy="Comment">
-            <div className="message-header">
-              <a href="mailto:misha@mate.academy" data-cy="CommentAuthor">
-                Misha Hrynko
-              </a>
+      setComments([...comments, commentDataFromServer]);
+    } catch {
+      setComments(comments);
+    }
+  };
 
-              <button
-                data-cy="CommentDelete"
-                type="button"
-                className="delete is-small"
-                aria-label="delete"
-              >
-                delete button
-              </button>
-            </div>
-            <div className="message-body" data-cy="CommentBody">
-              One more comment
-            </div>
-          </article>
-
-          <article className="message is-small" data-cy="Comment">
-            <div className="message-header">
-              <a href="mailto:misha@mate.academy" data-cy="CommentAuthor">
-                Misha Hrynko
-              </a>
+  return (
+    <div className="content" data-cy="PostDetails">
+      <div className="content" data-cy="PostDetails">
+        <div className="block">
+          <h2 data-cy="PostTitle">
+            {`#${selectedPost.id}: ${selectedPost.title}`}
+          </h2>
 
-              <button
-                data-cy="CommentDelete"
-                type="button"
-                className="delete is-small"
-                aria-label="delete"
-              >
-                delete button
-              </button>
+          <p data-cy="PostBody">{selectedPost.body}</p>
+        </div>
+        <div className="block">
+          {isError && (
+            <div className="notification is-danger" data-cy="CommentsError">
+              Something went wrong
             </div>
+          )}
 
-            <div className="message-body" data-cy="CommentBody">
-              {'Multi\nline\ncomment'}
-            </div>
-          </article>
+          {isLoading ? (
+            <Loader />
+          ) : comments.length === 0 && !isError ? (
+            <p className="title is-4" data-cy="NoCommentsMessage">
+              No comments yet
+            </p>
+          ) : (
+            <>
+              <p className="title is-4">Comments:</p>
+              {comments.map((comment, index) => (
+                <article
+                  className="message is-small"
+                  key={index}
+                  data-cy="Comment"
```

Comment:
It's a bad practice to use index as key
```suggestion
              {comments.map((comment) => (
                <article
                  className="message is-small"
                  key={comment.id}
                  data-cy="Comment"
```